eqcorrscan/utils/catalog_to_dd.py

```diff
@@ -235,7 +242,11 @@ def _compute_dt_correlations(catalog, master, min_link, event_id_mapper,
     Logger.debug(f"Using channels: {master_seed_ids}")
     # Dictionary of travel-times for master keyed by {station}_{phase_hint}
     master_tts = dict()
-    master_origin_time = (master.preferred_origin() or master.origins[0]).time
+    try:
+        master_origin_time = (
+            master.preferred_origin() or master.origins[0]).time
+    except AttributeError:  # In case it's a SparseEvent
```

Comment:
Maybe `SparseEvent` should have a `preferred_origin` method that returns a `SparseOrigin`.

Reply:
The thing is that we then have to add the `preferred_origin_id`-property and the function itself which makes the event a bit less sparse... not a problem if you prefer it, but I understood that the `SparseEvent` should maybe be as sparse as possible?

Reply:
Yeah, I think you are right - sparsity is good.